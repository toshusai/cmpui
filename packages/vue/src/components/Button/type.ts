export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
