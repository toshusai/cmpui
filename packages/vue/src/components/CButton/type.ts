export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
