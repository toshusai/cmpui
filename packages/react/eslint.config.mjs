"use strict";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";
import eslintPluginImport from "eslint-plugin-import";

export default [
  ...tseslint.configs.recommended,
  {
    ignores: ["dist/*", "node_modules/*", "storybook-static/*"],
  },
  {
    plugins: { import: eslintPluginImport },
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
      "import/order": [
        "error",
        {
          "newlines-between": "always",
        },
      ],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
