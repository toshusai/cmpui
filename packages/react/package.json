{
  "name": "@toshusai/cmpui",
  "version": "1.0.0-alpha.13",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "dist/index.mjs",
  "types": "dist/index.d.mts",
  "scripts": {
    "typecheck": "tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "dev": "tsup --watch",
    "build": "tsup",
    "storybook": "storybook dev -p 6007",
    "storybook:build": "storybook build",
    "storybook:test": "test-storybook",
    "storybook:test:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"pnpm storybook:build && npx http-server storybook-static --port 6006 --silent\" \"wait-on http://127.0.0.1:6006 && pnpm storybook:test\"",
    "format": "prettier .",
    "format:fix": "prettier --write .",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "fix": "npm run format:fix && npm run lint:fix"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.6.10",
    "@radix-ui/react-context-menu": "^2.1.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-menubar": "^1.0.4",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-toast": "^1.1.5",
    "@toshusai/cmpui-core": "link:../core",
    "@toshusai/cmpui-css": "link:../css",
    "react-transition-group": "^4.4.5"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.41.2",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/react": "8.2.6",
    "@storybook/react-vite": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@tabler/icons-react": "^3.15.0",
    "@types/jest": "^29.5.11",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@types/react-transition-group": "^4.4.11",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "8.2.6",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2",
    "vitest": "^1.1.0",
    "wait-on": "^7.2.0"
  },
  "keywords": [
    "react",
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
