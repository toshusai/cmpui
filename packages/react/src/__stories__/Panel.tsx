export const Panel = (
  props: {
    color: string;
    name: string;
  } & React.HTMLAttributes<HTMLDivElement>,
) => {
  return (
    <div
      style={{
        background: props.color,
        width: "100%",
        height: "100%",
        display: "flex",
        overflow: "auto",
        position: "relative",
      }}
      {...props}
    >
      <div
        style={{
          width: "100%",
          pointerEvents: "none",
          height: "100%",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          color: "white",
          fontSize: "32px",
          position: "absolute",
        }}
      >
        {props.name}
      </div>
      {/* <pre
        style={{
          fontSize: "12px",
          fontFamily: "monospace",
          margin: 0,
          padding: 8,
        }}
      >
        <code>{JSON.stringify(node?.node.toJson(), null, 4)}</code>
      </pre> */}
    </div>
  );
};
