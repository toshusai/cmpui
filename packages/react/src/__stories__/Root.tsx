export const Root = (props: { children: React.ReactNode }) => {
  return (
    <div
      style={{
        height: "512px",
        width: "100vw",
        display: "flex",
        padding: "4px",
        boxSizing: "border-box",
      }}
    >
      {props.children}
    </div>
  );
};
