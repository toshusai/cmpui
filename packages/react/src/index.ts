export * from "./components/Button";
export * from "./components/CanvasView";
export * from "./components/Checkbox";
export * from "./components/Circle";
export * from "./components/ColorInput";
export * from "./components/ColorLoupe";
export * from "./components/ColorPickerCircle";
export * from "./components/ColorPickerRect";
export * from "./components/ColorSlider";
export * from "./components/ContextMenu";
export * from "./components/FloatBox";
export * from "./components/Grid";
export * from "./components/HueCircle";
export * from "./components/IconButton";
export * from "./components/IconButton/IconButtonGroup";
export * from "./components/Line";
export * from "./components/ListItem";
export * from "./components/MenuBar/MenuBar";
export * from "./components/MenuBar/MenuBarButton";
export * from "./components/Modal";
export * from "./components/Popover";
export * from "./components/RectGizmo";
export * from "./components/Ruler";
export * from "./components/Select";
export * from "./components/SelectRect";
export * from "./components/Slider";
export * from "./components/SliderNumberField";
export * from "./components/SplitPane";
export * from "./components/SVPicker";
export * from "./components/Switch";
export * from "./components/TextArea";
export * from "./components/TextInput";
export * from "./components/Toast";
export * from "./components/Tooltip";
export * from "./components/TreeView";
export * from "./utils";
