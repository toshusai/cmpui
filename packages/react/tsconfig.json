{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "emitDeclarationOnly": true,
    "declaration": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "outDir": "dist",

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "typeRoots": ["node_modules/@types", ".storybook/@types"]
  },
  "include": ["src"]
}
