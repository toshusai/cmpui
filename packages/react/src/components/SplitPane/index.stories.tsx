import { useState } from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Panel } from "../../__stories__/Panel";
import { Root } from "../../__stories__/Root";

import { SplitPane } from ".";

const meta: Meta<typeof SplitPane> = {
  title: "Layout/SplitPane",
  component: SplitPane,
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof SplitPane>;

export const Horizontal: Story = {
  render: function Render() {
    const [sizes, setSizes] = useState<string[]>(
      Array(3).fill(`${(1 / 3) * 100}%`),
    );
    return (
      <Root>
        <SplitPane type="horizontal" sizes={sizes} onChangeRates={setSizes}>
          <Panel color="#a8e6cf" name="A" />
          <Panel color="#dcedc1" name="B" />
          <Panel color="#ffd3b6" name="C" />
        </SplitPane>
      </Root>
    );
  },
};

export const Readonly: Story = {
  render: function Render() {
    const [sizes] = useState<string[]>(Array(3).fill(`${(1 / 3) * 100}%`));
    return (
      <Root>
        <SplitPane type="horizontal" sizes={sizes}>
          <Panel color="#a8e6cf" name="A" />
          <Panel color="#dcedc1" name="B" />
          <Panel color="#ffd3b6" name="C" />
        </SplitPane>
      </Root>
    );
  },
};

export const MinMax: Story = {
  render: function Render() {
    const [sizes, setSizes] = useState<string[]>(
      Array(3).fill(`${(1 / 3) * 100}%`),
    );
    return (
      <Root>
        <SplitPane
          type="horizontal"
          sizes={sizes}
          onChangeRates={setSizes}
          minSizes={["100px", "auto", "10%"]}
          maxSizes={["200px", "auto", "50%"]}
        >
          <Panel color="#a8e6cf" name="A" />
          <Panel color="#dcedc1" name="B" />
          <Panel color="#ffd3b6" name="C" />
        </SplitPane>
      </Root>
    );
  },
};

export const Vertical: Story = {
  render: function Render() {
    const [sizes, setSizes] = useState<string[]>(
      Array(3).fill(`${(1 / 3) * 100}%`),
    );
    return (
      <Root>
        <SplitPane type="vertical" sizes={sizes} onChangeRates={setSizes}>
          <Panel key={0} color="#a8e6cf" name="A" />
          <Panel key={1} color="#dcedc1" name="B" />
          <Panel key={2} color="#ffd3b6" name="C" />
        </SplitPane>
      </Root>
    );
  },
};
