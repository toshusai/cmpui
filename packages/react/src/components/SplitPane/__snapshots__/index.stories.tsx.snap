// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/SplitPane Horizontal smoke-test 1`] = `
<div style="height: 512px; width: 100vw; display: flex; padding: 4px; box-sizing: border-box;">
  <div class="cmpui_split-pane__root"
       data-orientation="horizontal"
  >
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#a8e6cf"
           name="A"
           style="background: rgb(168, 230, 207); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          A
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#dcedc1"
           name="B"
           style="background: rgb(220, 237, 193); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          B
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#ffd3b6"
           name="C"
           style="background: rgb(255, 211, 182); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          C
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout/SplitPane MinMax smoke-test 1`] = `
<div style="height: 512px; width: 100vw; display: flex; padding: 4px; box-sizing: border-box;">
  <div class="cmpui_split-pane__root"
       data-orientation="horizontal"
  >
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: 100px; max-width: 200px;"
    >
      <div color="#a8e6cf"
           name="A"
           style="background: rgb(168, 230, 207); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          A
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#dcedc1"
           name="B"
           style="background: rgb(220, 237, 193); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          B
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: 10%; max-width: 50%;"
    >
      <div color="#ffd3b6"
           name="C"
           style="background: rgb(255, 211, 182); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          C
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout/SplitPane Readonly smoke-test 1`] = `
<div style="height: 512px; width: 100vw; display: flex; padding: 4px; box-sizing: border-box;">
  <div class="cmpui_split-pane__root"
       data-orientation="horizontal"
  >
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#a8e6cf"
           name="A"
           style="background: rgb(168, 230, 207); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          A
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="false"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#dcedc1"
           name="B"
           style="background: rgb(220, 237, 193); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          B
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="horizontal"
         data-resizable="false"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 33.3333%; height: 100%; min-width: auto;"
    >
      <div color="#ffd3b6"
           name="C"
           style="background: rgb(255, 211, 182); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          C
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout/SplitPane Vertical smoke-test 1`] = `
<div style="height: 512px; width: 100vw; display: flex; padding: 4px; box-sizing: border-box;">
  <div class="cmpui_split-pane__root"
       data-orientation="vertical"
  >
    <div class="cmpui_split-pane__child"
         style="width: 100%; height: 33.3333%; min-height: auto;"
    >
      <div color="#a8e6cf"
           name="A"
           style="background: rgb(168, 230, 207); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          A
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="vertical"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 100%; height: 33.3333%; min-height: auto;"
    >
      <div color="#dcedc1"
           name="B"
           style="background: rgb(220, 237, 193); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          B
        </div>
      </div>
    </div>
    <div class="cmpui_split-pane__divider"
         data-orientation="vertical"
         data-resizable="true"
    >
      <div>
      </div>
    </div>
    <div class="cmpui_split-pane__child"
         style="width: 100%; height: 33.3333%; min-height: auto;"
    >
      <div color="#ffd3b6"
           name="C"
           style="background: rgb(255, 211, 182); width: 100%; height: 100%; display: flex; overflow: auto; position: relative;"
      >
        <div style="width: 100%; pointer-events: none; height: 100%; display: flex; justify-content: center; align-items: center; color: white; font-size: 32px; position: absolute;">
          C
        </div>
      </div>
    </div>
  </div>
</div>
`;
